beacon_node/beacon_chain/src/events.rs

```diff
@@ -18,6 +21,53 @@ impl<T: EthSpec> EventHandler<T> for WebSocketSender<T> {
     }
 }
 
+pub struct ServerSentEvents<T: EthSpec> {
+    // Bus<> is itself Sync + Send.  We use Mutex<> here only because of the surrounding code does
+    // not enforce mutability statically (i.e. relies on interior mutability).
+    head_changed_queue: Arc<Mutex<Bus<SignedBeaconBlockHash>>>,
+    log: Logger,
+    _phantom: PhantomData<T>,
+}
+
+impl<T: EthSpec> ServerSentEvents<T> {
+    pub fn new(log: Logger) -> (Self, Arc<Mutex<Bus<SignedBeaconBlockHash>>>) {
+        let bus = Bus::new(T::slots_per_epoch() as usize);
+        let mutex = Mutex::new(bus);
+        let arc = Arc::new(mutex);
+        let this = Self {
+            head_changed_queue: arc.clone(),
+            log: log,
+            _phantom: PhantomData,
+        };
+        (this, arc)
+    }
+}
+
+impl<T: EthSpec> EventHandler<T> for ServerSentEvents<T> {
+    fn register(&self, kind: EventKind<T>) -> Result<(), String> {
+        match kind {
+            EventKind::BeaconHeadChanged {
+                current_head_beacon_block_root,
+                ..
+            } => {
+                let mut guard = self
+                    .head_changed_queue
+                    .lock()
+                    .map_err(|_| "Cannot lock mutex")?;
+                if let Err(_) = guard.try_broadcast(current_head_beacon_block_root.into()) {
+                    error!(
```

Comment:
Very minor, but this would be more consistent with other logs:

```rust
                    error!(
                        self.log,
                        "Head change streaming queue full";
                        "dropped_change" => format!("{}". current_head_beacon_block_root)
                    );
```